src/Service/Configurations/RuntimeConfigProvider.cs

```diff
@@ -5,407 +5,251 @@
+    /// Initialize the runtime configuration provider with the specified configurations.
+    /// This initialization method is used when the configuration is sent to the ConfigurationController
+    /// in the form of a string instead of reading the configuration from a configuration file.
+    /// </summary>
+    /// <param name="configuration">The engine configuration.</param>
+    /// <param name="schema">The GraphQL Schema. Can be left null for SQL configurations.</param>
+    /// <param name="connectionString">The connection string to the database.</param>
+    /// <param name="accessToken">The string representation of a managed identity access token</param>
+    /// <returns>true if the initialization succeeded, false otherwise.</returns>
+    public async Task<bool> Initialize(string jsonConfig, string? graphQLSchema, string connectionString, string? accessToken)
+    {
+        if (string.IsNullOrEmpty(connectionString))
         {
-            if (RuntimeConfiguration is null)
-            {
-                throw new DataApiBuilderException(
-                    message: "Runtime config isn't setup.",
-                    statusCode: HttpStatusCode.InternalServerError,
-                    subStatusCode: DataApiBuilderException.SubStatusCodes.ErrorInInitialization);
-            }
-
-            return RuntimeConfiguration;
+            throw new ArgumentException($"'{nameof(connectionString)}' cannot be null or empty.", nameof(connectionString));
         }
 
-        /// <summary>
-        /// Attempt to acquire runtime configuration metadata.
-        /// </summary>
-        /// <param name="runtimeConfig">Populated runtime configuartion, if present.</param>
-        /// <returns>True when runtime config is provided, otherwise false.</returns>
-        public virtual bool TryGetRuntimeConfiguration([NotNullWhen(true)] out RuntimeConfig? runtimeConfig)
+        if (string.IsNullOrEmpty(jsonConfig))
         {
-            runtimeConfig = RuntimeConfiguration;
-            return RuntimeConfiguration is not null;
+            throw new ArgumentException($"'{nameof(jsonConfig)}' cannot be null or empty.", nameof(jsonConfig));
         }
 
-        public virtual bool IsDeveloperMode()
-        {
-            return RuntimeConfiguration?.HostGlobalSettings.Mode is HostModeType.Development;
-        }
+        ManagedIdentityAccessToken = accessToken;
 
-        /// <summary>
-        /// Return whether to allow GraphQL introspection using runtime configuration metadata.
-        /// </summary>
-        /// <returns>True if introspection is allowed, otherwise false.</returns>
-        public virtual bool IsIntrospectionAllowed()
+        if (RuntimeConfigLoader.TryParseConfig(jsonConfig, out RuntimeConfig? runtimeConfig))
         {
-            return RuntimeConfiguration is not null && RuntimeConfiguration.GraphQLGlobalSettings.AllowIntrospection;
+            _runtimeConfig = runtimeConfig.DataSource.DatabaseType switch
```

Comment:
Can you add a comment explaining what is happening here, I was confused by the switch keyword but seemingly only one DatabaseType following.

Reply:
Wouldn't a comment potentially get stale/lost? It's leveraging the [`switch` expression](https://learn.microsoft.com/dotnet/csharp/language-reference/operators/switch-expression) combined with a [discard pattern](https://learn.microsoft.com/dotnet/csharp/language-reference/operators/patterns#discard-pattern), and the `_` is a [discard](https://learn.microsoft.com/dotnet/csharp/fundamentals/functional/discards) in C# that's used in a variety of locations in the codebase.